include/wx/generic/grid.h

```diff
@@ -1693,6 +1709,19 @@ class WXDLLIMPEXP_CORE wxGrid : public wxScrolledCanvas
     void RefreshBlock(int topRow, int leftCol,
                       int bottomRow, int rightCol);
 
+    // Refresh the whole area if _rect_ is NULL or just that rectangle otherwise
+    // (the rectangle is calculated in device coordinates)
+    //
+    // This function is more convenient/useful (from performance point of view)
+    // to use if the rectangle is already known/calculated, because:
+    // - it transparently handles frozen grid windows for us
+    // - it can deduce the labels area to be refreshed (if we asked to) from _rect_ which
+    //   represents the cells area. e.g. RefreshArea(wxGridArea::Cells | wxGridArea::Labels, ...);
+    // - there is no need to iterate over the selected blocks one more time to know which part of
+    //   the grid needs to be refreshed (for instance: transparent selection rely on this function
+    //   for performance reason)
+    void RefreshArea(int area, wxRect* rect = NULL, bool erasebg = true);
```

Comment:
Do we really need `erasebg` here? I think we can always erase it, at least I don't see any valid reason not to do it and it shouldn't really have any ill effects.

More importantly, does it really make sense to allow combining different areas _and_ specifying the rectangle? I.e. isn't it valid for just a single area only?

Reply:
> Do we really need `erasebg` here? I think we can always erase it, at least I don't see any valid reason not to do it and it shouldn't really have any ill effects.
> 

I agree with this and I'll remove it in the next commits

> More importantly, does it really make sense to allow combining different areas _and_ specifying the rectangle? I.e. isn't it valid for just a single area only?

I understand the confusion here, i.e:
One can effortlessly understand the meaning of the following calls:
```
RefreshArea(wxGA_Cells, &rect);
RefreshArea(wxGA_RowLabels, &rect);
RefreshArea(wxGA_ColLabels, &rect);
```
But the next ones are not !!
```
RefreshArea(wxGA_Labels, &rect); // wxGA_Labels means wxGA_RowLabels | wxGA_ColLabels 
RefreshArea(wxGA_Cells | wxGA_Labels, &rect);
```
What does `rect` represent here ? Well:
`RefreshArea(wxGA_Labels, &rect);` does not make sense and such call should be rejected or at least warn about it.

Whereas:
`RefreshArea(wxGA_Cells | wxGA_Labels, &rect);` means: `rect` is valid for `wxGA_Cells` area, and for the labels the rectangles are to be deduced/calculated from `rect`.

Maybe we should add more comments about the usage of this function to alleviate the confusion about the meaning of the parameter `rect` ?

Notice that the following calls are totally valid and shouldn't be there any confusion about them:
```
RefreshArea(wxGA_Labels); // refresh the entire wxGA_RowLabels area and wxGA_ColLabels area
RefreshArea(wxGA_Cells | wxGA_Labels); // as above, plus refresh the cells area too
```

Hope this is clear now ?

Reply:
I agree that having a function refreshing several areas entirely at once is convenient, but it seems that you also agree that using a rectangle and a combination of the areas doesn't make sense. So shouldn't there be 2 functions:

```cpp
void RefreshPartOfArea(wxGridArea area, const wxRect& r); // Refresh part of one area.
void RefreshArea(int areas); // Refresh one or more areas entirely.
```

This would be more clear IMO, what do you think?

Reply:
Yes this is what I've done now and I'll push the changes soon.